tests/functional/SynapseNext/SqlBuilderTest.php

```diff
@@ -0,0 +1,1192 @@
+                self::TEST_TABLE_IN_SCHEMA,
+                $timestampInit->format(DateTimeHelper::FORMAT)
+            )
+        );
+        $this->connection->exec(
+            sprintf(
+                'INSERT INTO %s([id],[col1],[col2],[_timestamp]) VALUES (1,\'2\',\'\',\'%s\')',
+                self::TEST_TABLE_IN_SCHEMA,
+                $timestampInit->format(DateTimeHelper::FORMAT)
+            )
+        );
+
+        $result = $this->connection->fetchAll(sprintf(
+            'SELECT * FROM %s',
+            self::TEST_TABLE_IN_SCHEMA
+        ));
+
+        $this->assertEqualsCanonicalizing([
+            [
+                'id' => '1',
+                'col1' => '',
+                'col2' => '1',
+                '_timestamp' => $timestampInit->format(DateTimeHelper::FORMAT) . '.000',
+            ],
+            [
+                'id' => '1',
+                'col1' => '2',
+                'col2' => '',
+                '_timestamp' => $timestampInit->format(DateTimeHelper::FORMAT) . '.000',
+            ],
+        ], $result);
+
+        // use timestamp
+        $options = new SynapseImportOptions(['col1'], false, true);
+        $sql = $this->getBuilder()->getUpdateWithPkCommand(
+            $fakeStage,
+            $fakeDestination,
+            $options,
+            $timestampSet->format(DateTimeHelper::FORMAT) . '.000'
+        );
+
+        $this->assertEquals(
+        // phpcs:ignore
+            'UPDATE [import-export-test-ng_schema].[import-export-test-ng_test] SET [col2] = COALESCE([src].[col2], \'\'), [_timestamp] = \'2020-01-01 01:01:01.000\' FROM [import-export-test-ng_schema].[#stagingTable] AS [src] WHERE [import-export-test-ng_schema].[import-export-test-ng_test].[col1] = COALESCE([src].[col1], \'\') AND (COALESCE(CAST([import-export-test-ng_schema].[import-export-test-ng_test].[col1] AS NVARCHAR(4000)), \'\') != COALESCE([src].[col1], \'\') OR COALESCE(CAST([import-export-test-ng_schema].[import-export-test-ng_test].[col2] AS NVARCHAR(4000)), \'\') != COALESCE([src].[col2], \'\')) ',
+            $sql
+        );
+        $this->connection->exec($sql);
+
+        $result = $this->connection->fetchAll(sprintf(
+            'SELECT * FROM %s',
+            self::TEST_TABLE_IN_SCHEMA
+        ));
+
+        foreach ($result as $item) {
+            $this->assertArrayHasKey('id', $item);
+            $this->assertArrayHasKey('col1', $item);
+            $this->assertArrayHasKey('col2', $item);
+            $this->assertArrayHasKey('_timestamp', $item);
+            $this->assertSame(
+                $timestampSet->format(DateTimeHelper::FORMAT),
+                (new DateTime($item['_timestamp']))->format(DateTimeHelper::FORMAT)
+            );
+        }
+    }
+
+    /**
+     * hpcs:disable SlevomatCodingStandard.TypeHints.TypeHintDeclaration
+     * @doesNotPerformAssertions
+     */
+    public function testTransaction(): void
```

Comment:
Tenhle test by chtělo v rámci dalšího tasku vylepšit. Když se zmení query v `getBeginTransaction` a `getCommitTransaction` tak to bude furt procházet. A stejně by se to mělo fixnout tam odkud je ten test zkopírovanej.

Reply:
upraveno